src/crypto/tls/conn.go

```diff
@@ -1475,6 +1475,7 @@ func (c *Conn) connectionStateLocked() ConnectionState {
 	state.SignedCertificateTimestamps = c.scts
 	state.OCSPResponse = c.ocspResponse
 	state.ECHAccepted = c.ech.accepted
+	state.ECHOffered = c.ech.offered
```

Comment:
`c.ech.offered == true` if ECH was offered by the client and rejected or accepted by the server.
`c.ech.greased == true` if the client sent GREASE ECH.

Hence, the condition you're after is `c.ech.offered || c.ech.greased`.

Reply:
Fixed.